src/ui/components/shared/WorkspaceSettingsModal/WorkspaceSettingsModal.tsx

```diff
@@ -221,24 +220,24 @@ const settings: Settings<
   },
 ];
 
-function WorkspaceSettingsModal({ view, ...rest }: PropsFromRedux) {
+export default function WorkspaceSettingsModal() {
+  const selectedTab = useAppSelector(state => {
+    const opts = selectors.getModalOptions(state);
+    const view = opts && "view" in opts ? opts.view : null;
+    return (
+      view &&
+      {
+        billing: "Billing",
+        members: "Team Members",
+        api: "API Keys",
+      }[view]
+    );
+  });
   const workspaceId = useGetTeamIdFromRoute();
-  const [selectedTab, setTab] = useState<string>();
   const { members } = hooks.useGetWorkspaceMembers(workspaceId);
   const { workspace } = hooks.useGetWorkspace(workspaceId);
   const { userId: localUserId } = hooks.useGetUserId();
 
-  useEffect(() => {
-    if (view) {
-      const views: Record<string, string> = {
-        billing: "Billing",
-        members: "Team Members",
-        api: "API Keys",
-      };
-      setTab(views[view]);
```

Comment:
This was trying to derive the local state from the selected state - so I've just refactored this to use the actual selected state right from the start